lib/client.js

```diff
@@ -80,6 +80,9 @@ function Client (options,readyCallback) {
       if (!self.requestExecutor.authToken) {
         return cb(new Error(strings.errors.NO_AUTH_TOKEN_HEADER));
       }
+
+      self.replaceJWTInUrl(self.requestExecutor.authToken);
```

Comment:
```javascript
27    self.jwt = opts.token || self.getJwtFromUrl();
```
If the original JWT comes from `opts` rather than the URL, I don't think it should store the new JWT in the URL? :)